src/main.rs

```diff
@@ -306,18 +306,27 @@ fn main() {
 
     // Set the title color if we're usign os colors
     if config.use_os_color {
-        let id: &str = if let Some(ref x) = args.distro_override {
-            x
+        let c: CrabFetchColor = if let Some(ref x) = args.distro_override {
+            formatter::find_os_color(x)
         } else {
             if known_outputs.os.is_none() {
                 let os_bench: Option<Instant> = benchmark_point(args.benchmark); 
                 known_outputs.os = Some(os::get_os(&config, &mut syscall_cache));
                 print_bench_time(args.benchmark, args.benchmark_warn, "OS (for OS Color)", os_bench);
             }
-            &known_outputs.os.as_ref().unwrap().as_ref().unwrap().distro_id
+            let mut temp: CrabFetchColor = formatter::find_os_color(&known_outputs.os.as_ref().unwrap().as_ref().unwrap().distro_id);
+            if temp == CrabFetchColor::Clear {
+                temp = formatter::find_os_color(&known_outputs.os.as_ref().unwrap().as_ref().unwrap().distro_id_like)
+            }
+
+            #[cfg(target_os = "linux")]
+            if temp == CrabFetchColor::Clear {
```

Comment:
You already checked this above, move this up into the same block

Reply:
This is to check `temp = formatter::find_os_color(&known_outputs.os.as_ref().unwrap().as_ref().unwrap().distro_id_like)`, not `distro_id`

Reply:
`if temp.0.is_empty() {` gets checked in both scenarios - Move `temp = formatter::find_os_color("linux")` up into the same block with your conditional compile so you only ever check it once - use the `cfg!` macro if you need

Reply:
`temp = formatter::find_os_color(&known_outputs.os.as_ref().unwrap().as_ref().unwrap().distro_id_like)` Can also fail. How can I check it if I remove the `if` check?